readthedocs/doc_builder/backends/sphinx.py

```diff
@@ -390,65 +376,62 @@ def __init__(self, *args, **kwargs):
 
 
 class LocalMediaBuilder(BaseSphinx):
-    type = 'sphinx_localmedia'
     sphinx_builder = 'readthedocssinglehtmllocalmedia'
-    sphinx_build_dir = '_build/localmedia'
-
-    @restoring_chdir
-    def move(self, **__):
-        if not os.path.exists(self.old_artifact_path):
-            log.warning('Not moving localmedia because the build dir is unknown.')
-            return
+    sphinx_build_dir = "_readthedocs/htmlzip"
 
-        log.debug('Creating zip file from path.', path=self.old_artifact_path)
+    def _post_build(self):
+        """Internal post build to create the ZIP file from the HTML output."""
+        sphinx_build_dir = os.path.join(self.project_path, self.sphinx_build_dir)
+        temp_zip_file = tempfile.mktemp(suffix=".zip")
         target_file = os.path.join(
-            self.target,
-            '{}.zip'.format(self.project.slug),
+            sphinx_build_dir,
+            f"{self.project.slug}.zip",
         )
-        if not os.path.exists(self.target):
-            os.makedirs(self.target)
-        if os.path.exists(target_file):
-            os.remove(target_file)
-
-        # Create a <slug>.zip file
-        os.chdir(self.old_artifact_path)
-        archive = zipfile.ZipFile(target_file, 'w')
-        for root, __, files in os.walk('.'):
+
+        archive = zipfile.ZipFile(temp_zip_file, "w")
+        for root, __, files in os.walk(sphinx_build_dir):
             for fname in files:
                 to_write = os.path.join(root, fname)
                 archive.write(
                     filename=to_write,
                     arcname=os.path.join(
-                        '{}-{}'.format(self.project.slug, self.version.slug),
-                        to_write,
+                        f"{self.project.slug}-{self.version.slug}",
+                        os.path.relpath(
+                            to_write,
+                            sphinx_build_dir,
+                        ),
                     ),
                 )
         archive.close()
 
+        safe_rmtree(sphinx_build_dir)
```

Comment:
This produces tests to fail with this exception:

```
  File "/usr/src/app/checkouts/readthedocs.org/readthedocs/doc_builder/backends/sphinx.py", line 407, in _post_build
    safe_rmtree(sphinx_build_dir)
  File "/usr/src/app/checkouts/readthedocs.org/readthedocs/core/utils/filesystem.py", line 146, in safe_rmtree
    _assert_path_is_inside_docroot(path)
  File "/usr/src/app/checkouts/readthedocs.org/readthedocs/core/utils/filesystem.py", line 32, in _assert_path_is_inside_docroot
    raise SuspiciousFileOperation(path)
django.core.exceptions.SuspiciousFileOperation: /tmp/readthedocs-tests/git-repository/_readthedocs/htmlzip
```

It's totally fine to delete that directory. @stsewd why we are getting this error here? I think this line should be inside the `if path.is_symlink()`, right? https://github.com/readthedocs/readthedocs.org/blob/main/readthedocs/core/utils/filesystem.py#L146

Reply:
During tests, you need to override the DOCROOT setting to point to your temp directory.

https://github.com/readthedocs/readthedocs.org/blob/84a149a57ccd9c1b5cd47bb285263a273124a199/readthedocs/config/tests/test_config.py#L696-L696